src/page-multi-facility/FacilityInputForm.tsx

```diff
@@ -44,14 +44,37 @@ const FacilityInputForm: React.FC = () => {
   const model = useModel();
 
   const save = () => {
+    // When saving an object to Firestore, Firestore will reject it if it is not
+    // serializable. This type enforces that what we are saving is serializable.
+    // In the future, we add new properties to the model that are not
+    // serializable, we will get a TS error below.
+    //
+    // Definition inspired by the one posted here:
+    // https://github.com/microsoft/TypeScript/issues/1897#issuecomment-338650717
+    type JsonSerializable =
+      | boolean
+      | number
+      | Date
+      | string
+      | undefined
+      | JsonSerializable[]
+      | { [key: string]: JsonSerializable };
+
+    let modelInputs: JsonSerializable = {
+      ...model[0],
+
+      // We don't need to save localeDataSource.
+      localeDataSource: undefined,
+    };
+
     saveFacility({
       id: facility?.id,
       name: facilityName || null,
       description: description || null,
       systemType: systemType || null,
-      modelInputs: JSON.parse(JSON.stringify(model))[0],
+      modelInputs,
```

Comment:
@szhu what you have here works in both IE11 and other browsers? What testing have we done?

Reply:
This seems to work in both IE11 and Chrome. (I can save a facility and recall it. Previously, I could not save the facility in IE11.) It appeared safe to omit `localeData` because even in Chrome, `JSON.stringify(localeData)` evaluates you `{}`, so I don't think it was being saved before.

I'm not as familiar as some you with what the correct functionality is or what exactly `localeData` is, so I plan on doing a call to test this out (like the one I had with @macfarlandian on Friday) before merging it.

Reply:
I think there's really no reason for us to be serializing `localeData` anyway — we shouldn't be saving it, that is the data structure we are loading and building from a separate datasource in Google Sheets. There is an enumerated list of `persistedKeys` in the `EpidemicModelContext` module and anything that's not in that list can be safely discarded (and indeed SHOULD be discarded) for persistence operations

Reply:
Okay cool thanks @macfarlandian! That's the kind of reassurance I was looking for 🙌

Reply:
We use `persistedKeys` to filter out values we're not interested in saving at the persistence layer:  https://github.com/Recidiviz/covid19-dashboard/blob/master/src/database/index.tsx#L313-L315

Reply:
@sprad and I resolved this on a call following the above advice. The `persistedKeys` correctly already filters out `localeDataSource`.